go/vt/vttablet/tabletmanager/rpc_external_reparent.go

```diff
@@ -98,23 +91,22 @@ func (agent *ActionAgent) TabletExternallyReparented(ctx context.Context, extern
 		},
 		ExternalID: externalID,
 	}
-	defer func() {
-		if err != nil {
-			event.DispatchUpdate(ev, "failed: "+err.Error())
-		}
-	}()
 	event.DispatchUpdate(ev, "starting external from tablet (fast)")
 
 	// Execute state change to master by force-updating only the local copy of the
 	// tablet record. The actual record in topo will be updated later.
-	log.Infof("fastTabletExternallyReparented: executing change callback for state change to MASTER")
 	newTablet := proto.Clone(tablet).(*topodatapb.Tablet)
-	newTablet.Type = topodatapb.TabletType_MASTER
 
-	// This is where updateState will block for gracePeriod, while it gives
-	// vtgate a chance to stop sending replica queries.
-	agent.updateState(ctx, newTablet, "fastTabletExternallyReparented")
+	// We may get called on the current master multiple times in order to fix incomplete external reparents.
+	// We update the tablet here only if it is not currently master
+	if newTablet.Type != topodatapb.TabletType_MASTER {
```

Comment:
It's ultimately the same behavior, but I think logically `newTablet` should be created inside this if block. We only need to make a clone of `tablet` if we're about to change it. This conditional will also be clearer since what we really want is to ask whether the old tablet is marked as MASTER or not.

Reply:
done